libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/Camera.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.services.android.navigation.v5.navigation.camera
+
+import com.mapbox.geojson.Point
+
+/**
+ * This class handles calculating camera's zoom and tilt properties while routing.
+ * The [com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation] uses
+ * a [SimpleCamera] by default. If you would like to customize the camera properties, create a
+ * concrete implementation of this class or subclass [SimpleCamera] and update
+ * [com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation.setCameraEngine].
+ *
+ * @since 0.10.0
+ */
+abstract class Camera {
```

Comment:
why not interface? wouldn't all abstract methods justify interface?

Reply:
This is how it was and don't know why not `interface`. In any case I am going to leave it like this and focus on just converting to kotlin. Once we pass all the tests and finish the `kotlin` conversion I would deep dive and see what can be done here